qemu/tests/virtio_scsi_mq.py

```diff
@@ -58,181 +58,198 @@ def proc_interrupts_results(results, irqs_pattern):
+        error_context.context("Check the interrupt queues in guest", logging.info)
+        irqs, _ = proc_interrupts_results(session.cmd_output(irq_check_cmd), irq_name)
+        logging.debug('The info of interrupters:')
+        for irq_watch in irqs.keys():
+            logging.debug(
+                '%s : %s %s' % (
+                    irq_watch, irqs[irq_watch]['count'], irqs[irq_watch]['irq_des']))
+        return irqs
+
+    def pin_interrupts2vcpus(irqs):
+        cpu_irq_map = {}
+        need_pin = True
+        if params.get('pin_interrupts2vcpus', 'yes') == 'no':
+            need_pin = False
+        else:
+            error_context.context("Pin the interrupters to vcpus", logging.info)
+        cpu_select = 1
+        for irq_id in irqs.keys():
+            if need_pin:
+                bind_cpu_cmd = "echo %s > /proc/irq/%s/smp_affinity" % (
+                    hex(cpu_select).replace('0x', ''), irq_id)
+            cpu_select = int(session.cmd_output(
+                        'cat /proc/irq/%s/smp_affinity' % irq_id))
+            cpu_irq_map[irq_id] = bin(cpu_select).replace('0b', '').count('0')
+            if need_pin:
+                cpu_select = cpu_select << 1
+                session.cmd(bind_cpu_cmd)
+        return cpu_irq_map
+
+    def load_io_data_disks():
+        error_context.context("Load I/O in all targets", logging.info)
+        output = session.cmd_output(params.get("get_dev_cmd", "ls /dev/[svh]d*"))
+        system_dev = re.findall(r"/dev/[svh]d\w+(?=\d+)", output)[0]
+        dd_timeout = int(re.findall(r"\d+", extra_image_size)[0])
+        fill_cmd = ""
+        count = 0
+        for dev in re.split(r"\s+", output):
+            if not dev:
+                continue
+            if not re.findall(system_dev, dev):
+                fill_cmd += " dd of=%s if=/dev/urandom bs=1M " % dev
+                fill_cmd += "count=%s oflag=direct &" % dd_timeout
+                count += 1
+        if count != images_num:
+            test.error("Disks are not all show up in system. Output "
+                       "from the check command: %s" % output)
+        # As Bug 1177332 exists, mq is not supported completely.
+        # So don't considering performance currently, dd_timeout is longer.
+        dd_timeout = dd_timeout * images_num * 2
+        session.cmd(fill_cmd, timeout=dd_timeout)
+
+        dd_thread_num = count
+        while dd_thread_num:
+            time.sleep(5)
+            output = session.cmd_output_safe("pgrep -lx dd", timeout=dd_timeout)
+            dd_thread_num = len(re.findall(r'\d+\s+dd$', output, re.M | re.I))
+
+    def compare_interrupts(prev_irqs, cur_irqs):
+        cpu_not_used = []
+        for irq in prev_irqs.keys():
+            cpu = int(cpu_irq_map[irq])
+            diff_val = int(
+                cur_irqs[irq]['count'][cpu]) - int(prev_irqs[irq]['count'][cpu])
+            if diff_val < int(int(re.findall(r"\d+", extra_image_size)[0]) * 0.5):
```

Comment:
Judging conditions are ok. 0.5 is uncertain every time.
line 220 \"if (int(count) - int(prev_irq_results[irq_watch]["count"][index])) > 0:\"
line 226 \"if 2 ** index & irq_bit_map != 2 ** index:\"